### PR TITLE
[shuffler] Refactor/simplify codegen interfaces

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: hail/src/main/scala/is/hail/services/shuffler/ShuffleClient.scala [hunk 1]
```diff
@@ -32,99 +32,125 @@ object ShuffleClient {
...
+  def getSocketOnWorker(): Socket = DeployConfig.get.socket("shuffler")
+
+  def create(cb: EmitCodeBuilder, shuffle: SCanonicalShufflePointerSettable): CompileTimeShuffleClient = {
+    val ts = shuffle.st.virtualType
+
+    val socket: Value[Socket] = cb.newField[Socket]("socket",
+      Code.invokeScalaObject0[Socket](CompileTimeShuffleClient.getClass, "getSocketOnWorker"))
+
+    val in: Value[InputBuffer] = cb.newField[InputBuffer]("ib", shuffleBufferSpec.buildCodeInputBuffer(socket.invoke[InputStream]("getInputStream")))
+    val out: Value[OutputBuffer] = cb.newField[OutputBuffer]("ib", shuffleBufferSpec.buildCodeOutputBuffer(socket.invoke[OutputStream]("getOuputStream")))
```
Comment (danking) [inline hail/src/main/scala/is/hail/services/shuffler/ShuffleClient.scala:45]: ouput => output

**Review by danking — DISMISSED** [review]

File: hail/src/main/scala/is/hail/expr/ir/EmitCodeBuilder.scala [hunk 2]
```diff
@@ -213,8 +214,19 @@ class EmitCodeBuilder(val emb: EmitMethodBuilder[_], var code: Code[Unit]) exten
...
+    val x = sc.asPCode
+    strValue(x.pt, x.code)
+  }
 
   // for debugging
-  def println(cString: Code[String]*) = this += Code._printlns(cString:_*)
+  def println(cString: Code[String]*) = this += Code._printlns(cString: _*)
+
+  def logInfo(cs: Code[String]*): Unit = {
+    this += Code.invokeScalaObject1[String, Unit](ExposedUtils.getClass, "logInfo", cs.reduce[Code[String]] { case (l, r) => (l.concat(r)) })
```
Comment (danking) [inline hail/src/main/scala/is/hail/expr/ir/EmitCodeBuilder.scala:226]: Hmm. This doesn't log with a timestamp the way everything else does:
```
2021-05-01 03:18:08 INFO  Worker$:59 - is.hail.backend.service.Worker cf72c1d0ab3d5d2fcb47ff629b6c081fbf56d2db
2021-05-01 03:18:08 INFO  Worker$:60 - running job 0 at root gs://danking/tmp/hail/query/ym6uea3E86ULctQ2pNqvrQcNXUGB41IYd5d2lQEqVgY= with scratch directory '/batch/4afdd6c00aab4183a9b80b7953496e7f'
2021-05-01 03:18:08 INFO  Tokens:20 - tokens found for namespaces {dking}
2021-05-01 03:18:08 INFO  Requester:35 - creating HttpClient
2021-05-01 03:18:08 INFO  tls:42 - ssl config file found at /batch/4afdd6c00aab4183a9b80b7953496e7f/secrets/ssl-config/ssl-config.json
2021-05-01 03:18:09 INFO  Requester:55 - request GET http://internal.hail/dking/memory/api/v1alpha/objects?q=gs%3A%2F%2Fdanking%2Ftmp%2Fhail%2Fquery%2Fym6uea3E86ULctQ2pNqvrQcNXUGB41IYd5d2lQEqVgY%3D%2Ff
2021-05-01 03:18:09 INFO  Requester:55 - request GET http://internal.hail/dking/memory/api/v1alpha/objects?q=gs%3A%2F%2Fdanking%2Ftmp%2Fhail%2Fquery%2Fym6uea3E86ULctQ2pNqvrQcNXUGB41IYd5d2lQEqVgY%3D%2Fcontexts
2021-05-01 03:18:10 INFO  Requester:65 - request GET http://internal.hail/dking/memory/api/v1alpha/objects?q=gs%3A%2F%2Fdanking%2Ftmp%2Fhail%2Fquery%2Fym6uea3E86ULctQ2pNqvrQcNXUGB41IYd5d2lQEqVgY%3D%2Ff response 200
2021-05-01 03:18:10 INFO  Requester:65 - request GET http://internal.hail/dking/memory/api/v1alpha/objects?q=gs%3A%2F%2Fdanking%2Ftmp%2Fhail%2Fquery%2Fym6uea3E86ULctQ2pNqvrQcNXUGB41IYd5d2lQEqVgY%3D%2Fcontexts response 200
2021-05-01 03:18:10 INFO  WorkerTimer$:41 - readInputs took 2151.926174 ms.
2021-05-01 03:18:10 INFO  Hail:28 - Running Hail version 0.2.65-cf72c1d0ab3d
2021-05-01 03:18:12 INFO  root:17 - RegionPool: initialized for thread 1: main
2021-05-01 03:18:12 INFO  root:132 - RegionPool: REPORT_THRESHOLD: 256.0K allocated (256.0K blocks / 0 chunks), regions.size = 3, 0 current java objects, 0 max java objects, thread 1: main
2021-05-01 03:18:12 INFO  root:132 - RegionPool: REPORT_THRESHOLD: 512.0K allocated (512.0K blocks / 0 chunks), regions.size = 7, 0 current java objects, 0 max java objects, thread 1: main
2021-05-01 03:18:12 INFO  Tokens:20 - tokens found for namespaces {dking}
2021-05-01 03:18:12 INFO  Requester:55 - request GET http://internal.hail/dking/address/api/shuffler
2021-05-01 03:18:12 INFO  Requester:65 - request GET http://internal.hail/dking/address/api/shuffler response 200
2021-05-01 03:18:12 INFO  services:133 - attempting to connect shuffler at 10.32.2.5:443
2021-05-01 03:18:12 INFO  services:137 - connected to shuffler at 10.32.2.5:443
shuffle get: start={_key: {idx: 150}}, startInclusive=true, end={_key: {idx: 185}}, endInclusive=false
operation 2 uuid /3WHfvQr+gN8k1k/iWOWRubxhDcw+UGGJzzcJKNR2IE=

```

Comment (tpoterba) [reply hail/src/main/scala/is/hail/expr/ir/EmitCodeBuilder.scala]: I think that's because I forgot to convert a few of the cb.println => cb.logInfo

**Comment by danking** [comment]
I'm good with this, but I want my service PR to merge first so that we can adjust the test failure annotations accordingly.

**Review by danking — APPROVED** [review]

**Comment by danking** [comment]
I might fail some xfails.

**Pull request merged**